satpy/composites/__init__.py

```diff
@@ -1082,28 +1103,33 @@ def _get_and_sharpen_rgb_data_arrays_and_meta(self, datasets, optional_datasets)
             if 'rows_per_scan' in high_res.attrs:
                 new_attrs.setdefault('rows_per_scan', high_res.attrs['rows_per_scan'])
             new_attrs.setdefault('resolution', high_res.attrs['resolution'])
-            low_res_colors = ['red', 'green', 'blue']
-            low_resolution_index = low_res_colors.index(self.high_resolution_color)
+
         else:
             LOG.debug("No sharpening band specified for ratio sharpening")
             high_res = None
-            low_resolution_index = 0
 
+        bands = {'red': low_res_red, 'green': low_res_green, 'blue': low_res_blue}
+        self._sharpen_bands_with_high_res(bands, high_res)
+
+        return bands['red'], bands['green'], bands['blue'], new_attrs
+
+    def _sharpen_bands_with_high_res(self, bands, high_res):
         if high_res is not None:
-            low_res = (low_res_red, low_res_green, low_res_blue)[low_resolution_index]
             ratio = da.map_blocks(
                 _get_sharpening_ratio,
                 high_res.data,
-                low_res.data,
+                bands[self.high_resolution_color].data,
                 meta=np.array((), dtype=high_res.dtype),
                 dtype=high_res.dtype,
                 chunks=high_res.chunks,
             )
+
+            bands[self.high_resolution_color] = high_res
+
             with xr.set_options(keep_attrs=True):
-                low_res_red = high_res if low_resolution_index == 0 else low_res_red * ratio
-                low_res_green = high_res if low_resolution_index == 1 else low_res_green * ratio
-                low_res_blue = high_res if low_resolution_index == 2 else low_res_blue * ratio
-        return low_res_red, low_res_green, low_res_blue, new_attrs
+                for color in bands.keys():
+                    if color != self.neutral_resolution_color and color != self.high_resolution_color:
+                        bands[color] = bands[color] * ratio
```

Comment:
I was thinking `if high_res is not None:` would stay outside the method. If you want it inside the method could you invert the condition and do:

```
if high_res is None:
    return
... high res code ...
```